src/eureka/S3_data_reduction/optspex.py

```diff
@@ -673,38 +640,49 @@ def optimize_wrapper(data, meta, log, apdata, apmask, apbg, apv0, gain=1,
     data['optmask'].attrs['time_units'] = data.flux.attrs['time_units']
     data['optmask'].attrs['wave_units'] = data.wave_1d.attrs['wave_units']
 
-    # Select median frame over aperture region
-    ap_y1 = int(meta.src_ypos-meta.spec_hw)
-    ap_y2 = int(meta.src_ypos+meta.spec_hw+1)
-    apmedflux = data.medflux[ap_y1:ap_y2].values
-
     # Perform optimal extraction on each of the frames
     iterfn = range(meta.int_start, meta.n_int)
     if meta.verbose:
         iterfn = tqdm(iterfn)
-    for n in iterfn:
-        data['optspec'][n], data['opterr'][n], _ = \
-            optimize(meta, apdata[n], apmask[n], apbg[n],
-                     data.stdspec[n].values, gain, apv0[n],
-                     p5thresh=meta.p5thresh,
-                     p7thresh=meta.p7thresh,
-                     fittype=meta.fittype,
-                     window_len=meta.window_len,
-                     deg=meta.prof_deg, windowtype=windowtype,
-                     n=n, m=m, meddata=apmedflux)
+    if meta.orders is None:
+        for n in iterfn:
+            data['optspec'][n], data['opterr'][n], _ = \
+                optimize(meta, apdata[n], apmask[n], apbg[n],
+                         data.stdspec[n].values, gain, apv0[n],
+                         p5thresh=meta.p5thresh,
+                         p7thresh=meta.p7thresh,
+                         fittype=meta.fittype,
+                         window_len=meta.window_len,
+                         deg=meta.prof_deg, windowtype=windowtype,
+                         n=n, m=m, meddata=apmedflux)
+    else:
+        norders = len(meta.orders)
+        for n in iterfn:
+            for k in range(norders):
+                data['optspec'][n, :, k], data['opterr'][n, :, k], _ = \
+                    optimize(meta, apdata[n, :, :, k], apmask[n, :, :, k], 
+                             apbg[n, :, :, k], data.stdspec[n, :, k].values, 
+                             gain, apv0[n, :, :, k],
+                             p5thresh=meta.p5thresh,
+                             p7thresh=meta.p7thresh,
+                             fittype=meta.fittype,
+                             window_len=meta.window_len,
+                             deg=meta.prof_deg, windowtype=windowtype,
+                             n=n, m=m, meddata=apmedflux[:, :, k], 
+                             order=meta.orders[k])
```

Comment:
A for loop with `norders = 0` doesn't execute and with `norders = 1` we would need to break backwards compatibility.  For v2.0, we could easily make it so that every observing mode had `order` as a dimension and then we would no longer need the `if` statement.

Reply:
If we can't do this now, we should add a task to remind us to do this later